### DISCUSSION TITLE
Is there a way to configure dependencies between resources?

### DISCUSSION THREAD

**Discussion by Sheludchenko** [aztfmod/terraform-azurerm-caf#1182] [Q&A]
I'm trying to use CAF module to deploy AKS cluster with User Assigned Managed Identities for cluster and kubelet. Everything seems to be very straightforward and it is but I bumped into an issue.

In my tfvar file I define the following resources:
1. Resource Group
2. Virtual Network with subnet for AKS cluster
3. Two User Assigned Managed Identities
4. Role assignment (cluster identity is assigned with Managed Identity Operator role over kubelet identity)
5. AKS cluster which is linked to all resources at the top except for role assignment

When I try to deploy I get an error that cluster identity doesn't have a right to manage kubelet identity. The issue happens because AKS cluster gets created before or in parallel with role assignment.

Is there a way to set explicit dependency between AKS and role assignment?

**Comment by piroinno** [comment]
This would have been great
